### ISSUE TITLE
Move ARAX to KG2.6.7

### ISSUE THREAD

**Issue by amykglen** [RTXteam/RTX#1540]
- [x] build a new KG2c from KG2.6.7 (available at http://kg2-6-7c.rtx.ai:7474/browser/)
- [x] load it into plover (available at http://kg2-6-7cplover.rtx.ai:9990)

Rebuild or edit and put copies in `/translator/data/orangeboard/databases/KG2.6.7` on arax.ncats.io:

- [x] configv2.json (should point to the new KG2/KG2c/plover)
        * note: saved this as `config_local.json`, since we want it to be used over `configv2.json` during testing
- [x] KG2c meta knowledge graph
- [x] NodeSynonymizer
- [x] KG2c sqlite
- [x] NGD database
- [x] 'slim' databases (used for Travis) @chunyuma/@finnagin
- [x] COHD database @chunyuma
- [x] refreshed DTD @chunyuma
- [x] DTD model @chunyuma
- [x] DTD database @chunyuma

**Note**: As databases are rebuilt, the new copy of `config_local.json` will need to be updated to point to their new paths.

Other things:

- [x] update code base
       * code changes will go in the `kg2integration` branch
- [x] test everything together (entire pytest suite should pass when using the new `config_local.json` - must set `force_local=True` in `ARAX_expander.py` to avoid using the old KG2 API)
        * tweaks to tests should go in the `kg2integration` branch
- [ ] update the test triples that go in some NCATS repo? @finnagin

**Comment by amykglen** [comment]
ok, I've gotten `expand` (except COHD/DTD) and `resultify` tests working with KG2.6.7 in the `kg2integration` branch. 

@chunyuma and @finnagin - these tests are failing in the `kg2integration` branch - would be great if you could go in and fix yours (should just be due to changes in the structure of the graph/different predicates, etc.):
```
FAILED test_ARAX_expand.py::test_cohd_expand_paired_concept_freq - AssertionError: assert (False or False or False)
FAILED test_ARAX_expand.py::test_cohd_expand_observed_expected_ratio - AssertionError: assert (False or False or False)
FAILED test_ARAX_expand.py::test_dtd_expand_1 - AssertionError: assert (False or False or False)
FAILED test_ARAX_filter_kg.py::test_error - AssertionError: assert 'OK' == 'ERROR'
FAILED test_ARAX_filter_kg.py::test_provided_by_filter - assert 0 > 0
FAILED test_ARAX_overlay.py::test_add_node_pmids - assert 0 > 0
FAILED test_ARAX_overlay.py::test_predict_drug_treats_disease_virtual - AssertionError: assert 'biolink:probably_treats' in Counter()
FAILED test_ARAX_overlay.py::test_predict_drug_treats_disease_attribute - assert 0 >= 1
FAILED test_ARAX_overlay.py::test_issue_832 - AssertionError: assert 'biolink:probably_treats' in Counter()
FAILED test_ARAX_standup_queries.py::test_chemical_substances_correlated_with_asthma_issue_18 - AssertionError: assert 0 > 0
FAILED test_ARAX_standup_queries.py::test_phenotypes_for_angel_shaped_phalango_epiphyseal_dysplasia_issue_33 - AssertionError: assert 0 > 0
FAILED test_ARAX_workflows.py::test_example_2 - assert 3 == 15
FAILED test_ARAX_workflows.py::test_clinical_overlay_example1 - AssertionError: assert 1 >= 2
```
note that to test you'll need to:
1. copy `arax.ncats.io:/translator/data/orangeboard/databases/KG2.6.7/config_local.json` to your local `RTX/code/` directory
2. locally change [this line](https://github.com/RTXteam/RTX/blob/master/code/ARAX/ARAXQuery/ARAX_expander.py#L66) in `ARAX_expander.py` to `force_local = True` (otherwise the KG2.5.2 API will be used)

then when you run the pytests all necessary KG2.6.7 databases will be downloaded.

(as a side note - maybe at some point we could move that `force_local=True` setting somewhere into `configv2.json`... seems like that'd be more convenient)

**Comment by amykglen** [comment]
ok - all databases have been rebuilt (except for DTD, but it's been 'refreshed'), the remaining piece is just fixing any failing tests in `kg2integration`! (@finnagin, @chunyuma)

**Comment by amykglen** [comment]
about the two standup queries that are failing:
* test_ARAX_standup_queries.py::test_chemical_substances_correlated_with_asthma_issue_18
    * in KG2.5.2c, there was conflation of asthma with bronchial diseases, broncheal spasm, allergic bronchospasm, etc. - those conflations are gone in KG2.6.7c, and this query no longer produces results
* test_ARAX_standup_queries.py::test_phenotypes_for_angel_shaped_phalango_epiphyseal_dysplasia_issue_33
    * in KG2.5.2c, there was conflation of angel-shaped phalango epiphyseal dysplasia with other concepts like MELNICK-NEEDLES SYNDROME and Multiple epiphyseal dysplasia - those conflations are gone in KG2.6.7c, and this query no longer produces results.

**Comment by chunyuma** [comment]
ok, @amykglen, I fixed most of tests associated with COHD/DTD except for `test_dtd_expand_1`. As you said, most of them failed due to  changes in the structure of the graph/different predicates. 

`test_dtd_expand_1` can be solved until the new DTD model is created (which might need two more days). The reason why the refreshed DTD databases can't solve it is because in kg2.5.2c, `Sotos Syndrome` used `MONDO:0007299` as preferred curie and `MONDO:0019349` is one of its synonyms. Now in kg2.6.7, `Sotos Syndrome` used `MONDO:0019349` as preferred curie. 
 
Additionally, I fixed `test_ARAX_workflows.py::test_example_2` by directly changing `assert len(message.results) == 15` to `assert len(message.results) > 0 `. I think it is not reasonable to check if a number of results are equal to a fixed value becaue it might be changed once we updated the KG2. 

In summary, now the following tests are fixed in the `kg2integration` branch :
```
test_ARAX_expand.py::test_cohd_expand_paired_concept_freq 
test_ARAX_expand.py::test_cohd_expand_observed_expected_ratio
test_ARAX_overlay.py::test_predict_drug_treats_disease_virtual
test_ARAX_overlay.py::test_predict_drug_treats_disease_attribute 
test_ARAX_overlay.py::test_issue_832
test_ARAX_workflows.py::test_example_2
```

**Comment by amykglen** [comment]
awesome, thanks @chunyuma!

that makes sense about `test_dtd_expand_1` - since it will fail until the new DTD is ready, is there maybe another test case you could use instead? or could we mark the test as `@pytest.mark.skip` for the meantime? (so that we don't have a failing test)

**Comment by chunyuma** [comment]
hi @amykglen, I marked this test as @pytest.mark.skip temporally.

**Comment by amykglen** [comment]
ok, I fixed the remaining 'fast' tests that were failing, so now `pytest -v` is passing when KG2.6.7 is all hooked up

so I think we're ready to roll 2.6.7 out to ARAX production (maybe during today's minihackathon?). 

to do that, I think steps are:
1. Merge `kg2integration` into `master`
1. Replace the master config file on araxconfig.rtx.ai with the new 2.6.7 config file
1. Roll `master` out to all endpoints
1. Delete each endpoint's `configv2.json`
1. Run the pytest suite on each endpoint

**Comment by amykglen** [comment]
are we good to close this issue? @finnagin did you get those test cases updated in whatever NCATS repo they're in?

**Comment by finnagin** [comment]
Yes, I think we are good to close.
